src/app.js

```diff
@@ -497,7 +497,47 @@ var run = function () {
 
   var offsetToLineColumnConverter = new OffsetToLineColumnConverter(compiler.event)
 
-  var transactionDebugger = new Debugger('#debugger', editor, compiler, executionContext.event, switchToFile, offsetToLineColumnConverter)
+  var contentToolAPI = {
+    offsetToLineColumn: (location, file) => {
+      return offsetToLineColumnConverter.offsetToLineColumn(location, file, compiler.lastCompilationResult)
+    }
+  }
+
+  var ethToolAPI = {
+    toWei: (value, unit) => {
+      return executionContext.web3().toWei(value, unit)
+    }
+  }
+
+  var compilerAPI = {
+    lastCompilationResult: () => {
+      return compiler.lastCompilationResult
+    }
+  }
+
+  var editorAPI = {
+    currentOpenedFile: () => {
+      return editor.getCacheFile()
+    },
+    addMarker: (range, css) => {
```

Comment:
Replace addMarker and removeMarker by `currentSourceLocation` - or should it perhaps be an event?